game/hud/src/widgets/Crafting/services/session/job.ts

```diff
@@ -0,0 +1,287 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * @Author: Mehuge (mehuge@sorcerer.co.uk)
+ * @Date: 2017-05-03 20:46:31
+ * @Last Modified by: Mehuge (mehuge@sorcerer.co.uk)
+ * @Last Modified time: 2017-05-20 23:47:18
+ */
+
+import { client, hasClientAPI } from 'camelot-unchained';
+import { Module } from 'redux-typed-modules';
+import { slash, isClient } from '../game/slash';
+import { Ingredient, InventoryItem, Recipe, Template, Message, VoxStatus } from '../types';
+
+export interface JobState {
+  loading: boolean;                   // Are we starting up?
+  vox: string;                        // Vox ID (if known)
+  status: string;                     // Vox status (if known)
+  ready: boolean;                     // Crafting complete? (Item Ready)  -- TODO Do we need this?
+  type: string;                       // What type of crafting are we doing?
+  started: string;                    // When last job started
+  endin: string;                      // How long until it ends
+  recipe: Recipe;                     // Selected Recipe
+  template: Template;                 // Selected Template (make job)
+  quality: number;                    // Desired quality
+  possibleIngredients: InventoryItem[];  // ingredients that can go in the vox
+  ingredients: Ingredient[];          // ingredients in the vox
+  name: string;                       // Item Name (make)
+  message: Message;                   // Last message from vox
+  count: number;                      // Number of items to make
+}
+
+const initialState = () : JobState => {
+  return {
+    vox: null,
+    status: 'unknown',
+    ready: false,
+    loading: false,
+    type: null,
+    started: null,
+    endin: null,
+    recipe: null,
+    template: null,
+    quality: undefined,
+    possibleIngredients: [],
+    ingredients: [],
+    name: null,
+    message: null,
+    count: undefined,
+  };
+};
+
+const module = new Module({
+  initialState: initialState(),
+  actionExtraData: () => {
+    return {
+      when: new Date(),
+    };
+  },
+});
+
+export const setJobType = module.createAction({
+  type: 'crafting/job/set-job',
+  action: (jobType: string) => {
+    return { jobType };
+  },
+  reducer: (s, a) => {
+    return Object.assign(s, { type: a.jobType });
+  },
+});
+
+export const setCount = module.createAction({
+  type: 'crafting/job/set-count',
+  action: (count: number) => {
+    return { count };
+  },
+  reducer: (s, a) => {
+    return Object.assign(s, { count: a.count });
+  },
+});
+
+export const setLoading = module.createAction({
+  type: 'crafting/job/set-loading',
+  action: (loading: boolean) => {
```

Comment:
That's not the same.  The difference is

`setLoading(true)` vs `setLoading({ loading: true })`